tests/test_unittest/test_gtc/test_dace.py

```diff
@@ -0,0 +1,126 @@
+# distribution for a copy of the license or check <https://www.gnu.org/licenses/>.
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from copy import deepcopy
+
+import dace
+import dace.sdfg.graph
+import networkx as nx
+import pytest
+
+from gt4py.backend.gtc_backend.defir_to_gtir import DefIRToGTIR
+from gt4py.definitions import BuildOptions
+from gt4py.frontend.gtscript_frontend import GTScriptFrontend
+from gtc.dace.dace_to_oir import convert
+from gtc.dace.nodes import HorizontalExecutionLibraryNode, VerticalLoopLibraryNode
+from gtc.dace.oir_to_dace import OirSDFGBuilder
+from gtc.gtir_to_oir import GTIRToOIR
+from gtc.passes.gtir_pipeline import GtirPipeline
+
+from ...test_integration.stencil_definitions import EXTERNALS_REGISTRY as externals_registry
+from ...test_integration.stencil_definitions import REGISTRY as stencil_registry
+
+
+def stencil_def_to_oir(stencil_def, externals):
+
+    build_options = BuildOptions(
+        name=stencil_def.__name__, module=__name__, rebuild=True, backend_opts={}, build_info=None
+    )
+    definition_ir = GTScriptFrontend.generate(
+        stencil_def, externals=externals, options=build_options
+    )
+    gtir = GtirPipeline(DefIRToGTIR.apply(definition_ir)).full()
+    return GTIRToOIR().visit(gtir)
+
+
+def edge_match(edge1, edge2):
+    edge1 = next(iter(edge1.values()))
+    edge2 = next(iter(edge2.values()))
+    try:
+        if edge1["src_conn"] is not None:
+            assert edge2["src_conn"] is not None
+            assert edge1["src_conn"] == edge2["src_conn"]
+        else:
+            assert edge2["src_conn"] is None
+        assert edge1["data"] == edge2["data"]
+        assert edge1["data"].data == edge2["data"].data
+    except AssertionError:
+        return False
+    return True
+
+
+def node_match(n1, n2):
+    n1 = n1["node"]
+    n2 = n2["node"]
+    try:
+        if not isinstance(
+            n1, (dace.nodes.AccessNode, VerticalLoopLibraryNode, HorizontalExecutionLibraryNode)
+        ):
+            raise TypeError
+        if isinstance(n1, dace.nodes.AccessNode):
+            assert isinstance(n2, dace.nodes.AccessNode)
+            assert n1.access == n2.access
+            assert n1.data == n2.data
+        elif isinstance(n1, VerticalLoopLibraryNode):
+            assert isinstance(n2, VerticalLoopLibraryNode)
+            assert n1.loop_order == n2.loop_order
+            assert n1.caches == n2.caches
+            assert len(n1.sections) == len(n2.sections)
+            for (interval1, he_sdfg1), (interval2, he_sdfg2) in zip(n1.sections, n2.sections):
+                assert interval1 == interval2
+                assert_sdfg_equal(he_sdfg1, he_sdfg2)
+        else:
+            assert isinstance(n2, HorizontalExecutionLibraryNode)
+            assert n1.as_oir() == n2.as_oir()
+    except AssertionError:
+        return False
+    return True
+
+
+def assert_sdfg_equal(sdfg1: dace.SDFG, sdfg2: dace.SDFG):
```

Comment:
we've had problems with serializing (well, mainly deserializing, but still) before, also, how do you deal with isomorphic graphs then? the serialization can't always be the same, right?

Reply:
In our CI, we ensure determinism by file-comparing the SDFGs after a save-load-save roundtrip. In this case, it goes through another representation, which may shuffle node/edge ids etc. so graph isomorphism makes sense. However, the node-based comparison can still use a serialized string comparison, no?

Reply:
I guess it might work, but what's the benefit, expecially if just comparing nodes? The current state uses the __eq__methods which should be just fine, and forward to existing __eq__ code in oir.